ambari-server/src/main/java/org/apache/ambari/server/agent/RecoveryConfigHelper.java

```diff
@@ -314,6 +314,11 @@ public AutoStartConfig(String clusterName)
         return enabledComponents;
       }
 
+      // If the cluster has disabled recovery, there should be no service components can be auto started
+      if (configProperties.containsKey(RecoveryConfigHelper.RECOVERY_ENABLED_KEY) && configProperties.get(RECOVERY_ENABLED_KEY).equals("false")) {
```

Comment:
Maybe make this `if (configProperties.containsKey(...) && !BooleanUtils.isBoolean(str));`.  Using the hard-coded `"false"` may not always be the case unless you lower-case it (but please don't do that).

Reply:
thanks, I will change it.